Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/lucene/impl/LuceneUtility.java

```diff
@@ -0,0 +1,302 @@
+        } catch (ParseException e) {
+            throw new LuceneException("Issues in framing the query", e);
+        }
+    }
+
+    protected void createDocuments(Iterator<IIndexWord> iterator, IDictionary dict, IndexWriter writer)
+            throws IOException {
+        for (; iterator.hasNext();) {
+            IIndexWord indexWord = iterator.next();
+            List<IWordID> wordIdds = indexWord.getWordIDs();
+
+            for (IWordID wordId : wordIdds) {
+
+                Document doc = new Document();
+                doc.add(new StringField(LuceneFieldNames.WORD, wordId.getLemma(), Field.Store.YES));
+                doc.add(new StringField(LuceneFieldNames.POS, wordId.getPOS().toString(), Field.Store.YES));
+
+                IWord word = dict.getWord(wordId);
+                doc.add(new StringField(LuceneFieldNames.DESCRIPTION, word.getSynset().getGloss(), Field.Store.YES));
+                doc.add(new StringField(LuceneFieldNames.ID, word.getID().toString(), Field.Store.YES));
+                doc.add(new StringField(LuceneFieldNames.WORDNETID, word.getID().toString(), Field.Store.YES));
+
+                ISynset synset = word.getSynset();
+                List<IWord> synonyms = synset.getWords();
+                StringBuffer sb = new StringBuffer();
+                for (IWord syn : synonyms) {
+                    if (!syn.getID().equals(word.getID()))
+                        sb.append(
+                                syn.getID().toString() + edu.asu.conceptpower.servlet.core.Constants.SYNONYM_SEPARATOR);
+                }
+                doc.add(new StringField(LuceneFieldNames.SYNONYMID, sb.toString(), Field.Store.YES));
+                // Adding this new data to delete only wordnet concepts while
+                // adding all wordnet concepts from jwi.
+                doc.add(new StringField(LuceneFieldNames.CONCEPTTYPE, "wordnetconcept", Field.Store.YES));
+                writer.addDocument(doc);
+            }
+
+        }
+    }
+
+    @Override
+    public void indexConcepts() throws LuceneException {
+
+        try {
+            String wnhome = configuration.getWordnetPath();
+            String path = wnhome + File.separator + configuration.getDictFolder();
+
+            URL url = null;
+
+            url = new URL("file", null, path);
+
+            IDictionary dict = new Dictionary(url);
+            dict.open();
+
+            // 2. Adding data into
+            Iterator<IIndexWord> iterator = dict.getIndexWordIterator(POS.NOUN);
+            createDocuments(iterator, dict, writer);
+
+            iterator = dict.getIndexWordIterator(POS.ADVERB);
+            createDocuments(iterator, dict, writer);
+
+            iterator = dict.getIndexWordIterator(POS.ADJECTIVE);
+            createDocuments(iterator, dict, writer);
+
+            iterator = dict.getIndexWordIterator(POS.VERB);
+            createDocuments(iterator, dict, writer);
+
+            writer.commit();
```

Comment:
I'd commit the writer after every call to createdocuments to reduce memory usage